build.gradle.kts

```
@@ -14,6 +14,8 @@ dependencies {
     testImplementation(kotlin("test"))
     implementation("org.jetbrains.kotlinx:kotlinx-cli:0.3.5")
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.1")
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.5.1")
+
```

You should use [ktor](https://ktor.io/docs/getting-started-ktor-client.html) to make HTTP requests instead of Java's URL class. Have a look at their documentation, they include detailed excamples on how to make HTTP requests and deserialize content.

```suggestion
    implementation("io.ktor:ktor-client-core:2.3.0")
    implementation("io.ktor:ktor-client-cio:2.3.0")
```